diff --git a/generic/vendor/common/file.te b/generic/vendor/common/file.te
index 5cd1b156..37f93d15 100644
--- a/generic/vendor/common/file.te
+++ b/generic/vendor/common/file.te
@@ -273,4 +273,5 @@ type vendor_sysfs_microdump, fs_type, sysfs_type;
 type vendor_sysfs_disp_boot_param, fs_type, sysfs_type;
 
 #rmnet module files
-type vendor_sysfs_rmnet, fs_type, sysfs_type;
\ No newline at end of file
+type vendor_sysfs_rmnet, fs_type, sysfs_type;
+type vendor_sysfs_kgsl_max_gpuclk, sysfs_type, fs_type;
diff --git a/generic/vendor/common/file_contexts b/generic/vendor/common/file_contexts
index 58e0da99..c052642a 100644
--- a/generic/vendor/common/file_contexts
+++ b/generic/vendor/common/file_contexts
@@ -634,3 +634,5 @@
 
 # Microdump collector parameters
 /sys/module/microdump_collector/parameters/.*  u:object_r:vendor_sysfs_microdump:s0
+
+/sys/devices(/platform)?/soc/[a-f0-9]+.qcom,kgsl-3d0/kgsl/kgsl-3d0/max_gpuclk u:object_r:vendor_sysfs_kgsl_max_gpuclk:s0
